ntbs-service/DataMigration/NotificationImportService.cs

```diff
@@ -0,0 +1,126 @@
+                             INotificationRepository notificationRepository,
+                             NtbsContext context)
+        {
+            this.notificationSearcher = notificationSearcher;
+            this.notificationRepository = notificationRepository;
+            _context = context;
+        }
+
+        public async Task<IList<Notification>> ImportNotificationsAsync(string notificationId)
+        {
+            LogInformation($"Request to import Notification with Id={notificationId}");
+
+            if (notificationRepository.NotificationWithLtbrOrEtsIdExists(notificationId)) 
+            {
+                LogInformation($"Notification with Id={notificationId} already exists in database");
+                return null;
+            }
+
+            var notificationTask = notificationSearcher.Search(notificationId);
+            Task.WaitAll(notificationTask);
+            var notifications = notificationTask.Result.ToList();
+
+            LogInformation($"{notifications.Count} notifications included linked ones found to import");
+            var notificationsToSave = new List<Notification>();
+            foreach (var notification in notifications)
+            {
+                var linkedNotificationId = notification.LTBRID ?? notification.ETSID;  
+                LogInformation($"Validating notification with Id={linkedNotificationId}");
+
+                var validationErrors = GetValidationErrors(notification);
+                
+                if (validationErrors.Count() == 0)
+                {
+                    notificationsToSave.Add(notification);
+                    LogInformation($"No validation errors found");
+                }
+                else
+                {
+                    LogInformation($"{validationErrors.Count()} validation errors found:");
+                    foreach (var validationError in validationErrors)
+                    {
+                        LogInformation(validationError.ErrorMessage);
+                    }
+                }
+            }
+            LogInformation($"Importing {notificationsToSave.Count()} valid notifications");
+            var savedNotificaitons = await AddLinkedNotificationsAsync(notificationsToSave);
+
+
+            LogInformation($"Finished importing notification with Id={notificationId} and linked notifications");
+            return savedNotificaitons;
+        }
+
+        private IEnumerable<ValidationResult> GetValidationErrors(Notification notification) 
+        {
+            var validationsResults = new List<ValidationResult>();
+
+            notification.ShouldValidateFull = true;  
+            notification.PatientDetails.ShouldValidateFull = true;  
+            notification.ClinicalDetails.ShouldValidateFull = true;  
+            notification.TravelDetails.ShouldValidateFull = true;  
+            notification.VisitorDetails.ShouldValidateFull = true;  

```

Comment:
I think we have a piece of code which does this for all models in already - should we reuse that here?
https://github.com/PublicHealthEngland/ntbs_Beta/blob/528ec63da5d803a4865fc5da577fca80225d3fb1/ntbs-service/Pages/Notifications/NotificationEditModelBase.cs#L114

Reply:
SetShouldValidateFull requires notificationStatus which I thought was irrelevant when importing notification